src/draw/background.rs

```diff
@@ -1,25 +1,45 @@
-use raqote::Point;
+use raqote::{Point, SolidSource};
 
-use super::{DrawTarget, Drawable, Space};
-use crate::Color;
+use super::{DrawTarget, Drawable, RoundedRect, Space};
+use crate::{style::Radius, Color};
 
 pub struct Params {
+    pub width: u32,
+    pub height: u32,
     pub color: Color,
+    pub radius: Radius,
 }
 
-pub struct Background<'a> {
-    params: &'a Params,
+pub struct Background {
+    rect: RoundedRect,
 }
 
-impl<'a> Background<'a> {
-    pub fn new(params: &'a Params) -> Self {
-        Self { params }
+impl Background {
+    pub fn new(params: &Params) -> Self {
+        let color = params.color;
+        let radius = params.radius.clone();
+
+        Self {
+            rect: RoundedRect::new(radius, color),
+        }
     }
 }
 
-impl Drawable for Background<'_> {
-    fn draw(self, dt: &mut DrawTarget<'_>, _: u16, _: Space, _: Point) -> Space {
-        dt.clear(self.params.color.as_source());
+impl Drawable for Background {
+    fn draw(self, dt: &mut DrawTarget<'_>, scale: u16, space: Space, start_point: Point) -> Space {
+        // dbg!(scale);
+        let Space { width, height } = space;
+
+        // Clear the draw target to avoid artefacts for scales > 1 in the corners
+        dt.clear(SolidSource {
+            r: 0,
+            g: 0,
+            b: 0,
+            a: 0,
+        });
+
+        self.rect
+            .draw(dt, scale, Space { width, height }, start_point);
```

Comment:
What's the point of the destructuring assignment earlier for immediate constructing the exact same type here?

Reply:
There is none (anymore), fixed.